### PR TITLE
Used a Map to store sub-groups as opposed to a Set.

### PR DESCRIPTION
**Pull request by jmlvanre**

### CODE REVIEW

**Comment by jasongilanfarr** [comment]
LGTM. :+1:

**Comment by jdef** [comment]
lgtm

File: src/main/scala/mesosphere/marathon/state/Group.scala [hunk 1]
```diff
@@ -17,7 +17,7 @@ import scala.collection.JavaConversions._
 case class Group(
     id: PathId,
     apps: Map[AppDefinition.AppKey, AppDefinition] = defaultApps,
-    groups: Set[Group] = defaultGroups,
+    groupsById: Map[Group.GroupKey, Group] = defaultGroups,
```
Comment (aquamatthias) [inline src/main/scala/mesosphere/marathon/state/Group.scala]: Both apps and groups are available via id. Shouldn't we have the same scheme? (rename apps to appsById and have  a method to get apps)

**Comment by aquamatthias** [comment]
@jmlvanre Thanks looks good. Please rebase.

File: src/main/scala/mesosphere/marathon/state/Group.scala (hunk 1 above)
Comment (jmlvanre) [inline src/main/scala/mesosphere/marathon/state/Group.scala]: I agree. I think this should be a separate follow up commit.

**Comment by jasongilanfarr** [comment]
retest this please

**Pull request merged**